examples/jsm/utils/GeometryCompressionUtils.js

```diff
@@ -0,0 +1,812 @@
+
+        }
+
+
+
+
+    },
+
+
+    EncodingFuncs: {
+
+        defaultEncode: function (x, y, z, bytes) {
+            if (bytes == 1) {
+                let tmpx = Math.round((x + 1) * 0.5 * 255);
+                let tmpy = Math.round((y + 1) * 0.5 * 255);
+                let tmpz = Math.round((z + 1) * 0.5 * 255);
+                return new Uint8Array([tmpx, tmpy, tmpz]);
+            } else if (bytes == 2) {
+                let tmpx = Math.round((x + 1) * 0.5 * 65535);
+                let tmpy = Math.round((y + 1) * 0.5 * 65535);
+                let tmpz = Math.round((z + 1) * 0.5 * 65535);
+                return new Uint16Array([tmpx, tmpy, tmpz]);
+            } else {
+                console.error("number of bytes must be 1 or 2");
+            }
+        },
+
+        defaultDecode: function (array, bytes) {
+            if (bytes == 1) {
+                return [
+                    ((array[0] / 255) * 2.0) - 1.0,
+                    ((array[1] / 255) * 2.0) - 1.0,
+                    ((array[2] / 255) * 2.0) - 1.0,
+                ]
+            } else if (bytes == 2) {
+                return [
+                    ((array[0] / 65535) * 2.0) - 1.0,
+                    ((array[1] / 65535) * 2.0) - 1.0,
+                    ((array[2] / 65535) * 2.0) - 1.0,
+                ]
+            } else {
+                console.error("number of bytes must be 1 or 2");
+            }
+
+        },
+
+        // for `Angles` encoding
+        anglesEncode: function (x, y, z) {
+            let normal0 = parseInt(0.5 * (1.0 + Math.atan2(y, x) / Math.PI) * 65535);
+            let normal1 = parseInt(0.5 * (1.0 + z) * 65535);
+            return new Uint16Array([normal0, normal1]);
+        },
+
+        // for `OCT` encoding
+        octEncodeBest: function (x, y, z, bytes) {
+            var oct, dec, best, currentCos, bestCos;
+
+            // Test various combinations of ceil and floor
+            // to minimize rounding errors
+            best = oct = octEncodeVec3(x, y, z, "floor", "floor");
+            dec = octDecodeVec2(oct);
+            currentCos = bestCos = dot(x, y, z, dec);
+
+            oct = octEncodeVec3(x, y, z, "ceil", "floor");
+            dec = octDecodeVec2(oct);
+            currentCos = dot(x, y, z, dec);
+
+            if (currentCos > bestCos) {
+                best = oct;
+                bestCos = currentCos;
+            }
+
+            oct = octEncodeVec3(x, y, z, "floor", "ceil");
+            dec = octDecodeVec2(oct);
+            currentCos = dot(x, y, z, dec);
+
+            if (currentCos > bestCos) {
+                best = oct;
+                bestCos = currentCos;
+            }
+
+            oct = octEncodeVec3(x, y, z, "ceil", "ceil");
+            dec = octDecodeVec2(oct);
+            currentCos = dot(x, y, z, dec);
+
+            if (currentCos > bestCos) {
+                best = oct;
+                bestCos = currentCos;
```

Comment:
It seems the value of `bestCos` is never read, see https://lgtm.com/projects/g/mrdoob/three.js/snapshot/8ff9714090676d3f766f88d95e8c841b4250d541/files/examples/jsm/utils/GeometryCompressionUtils.js#x3ada8d6e7c54300f:1. It seems `octEncodeBest()` can be refactored to avoid this lgtm issue.